### PR TITLE
Fixes #4624 - Handle pasting text using context menu

### LINKED ISSUE #4624: [Annotations] The text background wrongly exceeds the pasted text if it contains paragraphs or a list
[Affected versions]:
- Nightly 63.0a1
- Dev Edition 62.0b7
- Release Unbranded

[Affected Platforms]:
- All Windows
- All Mac
- All Linux

[Prerequisites]:
- Have a clean Firefox profile with the latest Firefox Screenshots v33.0.0dev installed.
- Have a saved shot link.
- Have a text that contains at least one paragraph or a list, copied to the clipboard (eg. "This is just a test:
Test 1
Test 2").

[Steps to reproduce]:
1. Open the Firefox browser with the profile from prerequisites and navigate to the saved shot link.
2. Click the "Edit this image" button.
3. Click the "Add text" button and right click the text field.
4. Select the "Paste" option and click the "Confirm" button.
5. Observe the image.

[Expected result]:
- The pasted text is displayed on the image exactly the same way it was in the preview.

[Actual result]:
- The added text background exceeds the pasted text.

[Notes]:
- This issue is also reproducible if you paste a list.
- This issue is not reproducible if you paste an empty list or empty paragraphs.
- This issue is not reproducible if you paste the text using the "Ctrl/Cmd + V" keys combination.
- For the moment, we have a limitation: only a single line of text is allowed per text tool edit.
- Here is a screen recording with this issue:

ss - background

### PR DESCRIPTION
**Pull request by punamdahiya**

### CODE REVIEW

**Comment by punamdahiya** [comment]
@chenba Please review PR that handles pasting text inside text input via context menu. PR handles line break seen in copied text but remove any text inside p,list elements.  

There are couple of issues related to pasting text, #4630, #4628, #4627, #4625. PR might fix some of these but I will test and coment individually in respective issues

File: server/src/pages/shot/text-tool.js [hunk 1]
```diff
@@ -231,6 +232,17 @@ exports.TextTool = class TextTool extends React.Component {
     }
   }
 
+  onPaste(e) {
+    window.setTimeout(() => {
```
Comment (chenba) [inline server/src/pages/shot/text-tool.js]: Why a `setTimeout`?

File: server/src/pages/shot/text-tool.js [hunk 2]
```diff
@@ -231,6 +232,17 @@ exports.TextTool = class TextTool extends React.Component {
...
+      // Remove element tags, line breaks and new line seen after pasting text
+      while (this.textInput.current.firstElementChild) {
+        this.textInput.current.removeChild(this.textInput.current.firstElementChild);
+      }
+      this.textInput.current.textContent =  this.textInput.current.textContent.replace(/\r?\n|\r/g, "");
```
Comment (chenba) [inline server/src/pages/shot/text-tool.js]: eslint doesn't complain about two spaces after the equal sign in an assignment eh?

File: server/src/pages/shot/text-tool.js [hunk 3]
```diff
@@ -102,9 +103,9 @@ exports.TextTool = class TextTool extends React.Component {
 
     return [
       <div key="drag" style={dragDivStyles}
-        onMouseDown={this.onDragMouseDown.bind(this)} onMouseUp={this.onMouseUp.bind(this)} onMove={this.onMouseMove.bind(this)}>
+        onMouseDown={this.onDragMouseDown.bind(this)} onMouseUp={this.onMouseUp.bind(this)} onMouseMove={this.onMouseMove.bind(this)}>
```
[the remark this answers is not in the document]
Comment (punamdahiya) [reply server/src/pages/shot/text-tool.js]: ah ok will remove

File: server/src/pages/shot/text-tool.js (hunk 1 above)
Comment (punamdahiya) [reply server/src/pages/shot/text-tool.js]: that's the recommendation to delay execution of script tp ensure paste event has completed updating the textContent

File: server/src/pages/shot/text-tool.js (hunk 2 above)
Comment (punamdahiya) [reply server/src/pages/shot/text-tool.js]: nope, it didn't . good catch

**Comment by punamdahiya** [comment]
@chenba marking it on hold to fix a shift issue seen when pasted text is larger than canvas

File: server/src/pages/shot/text-tool.js [hunk 4]
```diff
@@ -52,6 +52,7 @@ exports.TextTool = class TextTool extends React.Component {
   componentDidMount() {
     this.textInput.current.focus();
     previousTextInputWidth = this.textInput.current.clientWidth;
+    previousInputText = "";
```
Comment (chenba) [inline server/src/pages/shot/text-tool.js]: What's the difference between setting the value here vs when the variable was declared?

Comment (punamdahiya) [reply server/src/pages/shot/text-tool.js]: Trying to recall, it was done to reset previousInputText for the next edit

**Comment by punamdahiya** [comment]
@chenba I am removing hold to keep shift fix separate, the second commit however fixes #4629

**Comment by punamdahiya** [comment]
Keeping commit separate for easy review to see what changed, will squash once approved

**Comment by chenba** [comment]
@punamdahiya Works great!

**Pull request merged**